jcl/src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -62,6 +62,11 @@ public boolean waitForReferenceProcessing() throws InterruptedException {
 	 */
 	static private native boolean waitForReferenceProcessingImpl();
 	
+	/* jdk.lang.ref.disableClearBeforeEnqueue property allow reverting to the old behavior(non clear before enqueue) 
+	   defer initializing the immutable variable to avoid bootstrap error */
+	static class disableClearBeforeEnqueue {
+		static final boolean ON = Boolean.getBoolean("jdk.lang.ref.disableClearBeforeEnqueue");
```

Comment:
Would `ENABLED` be a clearer name?  Or to remove the double negation, call the class `ClearBeforeEnqueue` and the variable `ENABLED` so the test can be `if (ClearBeforeEnqueue.ENABLED)`.

The return from getBoolean would have to be negated then as well